### PR TITLE
chore(docs): update preprocessing tutorial for using inst.pick() instead of pick_types()

### LINKED ISSUE #12158: Small error in interpolation tutorial
### Proposed documentation enhancement

https://mne.tools/dev/auto_tutorials/preprocessing/15_handling_bad_channels.html#interpolation-in-mne-python

Says:

> Note that we used the exclude=[] trick in the call to [pick_types()](https://mne.tools/dev/generated/mne.io.Raw.html#mne.io.Raw.pick_types) to make sure the bad channels were not automatically dropped from the selection.

But `pick_types()` isn't used anymore (we switched to `inst.pick()`)

### PR DESCRIPTION
**Pull request by btkcodedev**
#### Reference issue
Closes #12158

Updated the tutorial by removing legacy method call of mne.io.Raw.pick_types() to instance.pick()

### CODE REVIEW

File: tutorials/preprocessing/15_handling_bad_channels.py
Comment (wmvanvliet) [inline tutorials/preprocessing/15_handling_bad_channels.py:242]:
```suggestion
# :meth:`~mne.io.Raw.pick()` to make sure the bad channels were not
```

The `~` sign makes it so the `mne.io.Raw.` part doesn't show in the docs, but is present in the hyperlink.

**Review by wmvanvliet — CHANGES_REQUESTED** [review]

**Comment by wmvanvliet** [comment]
thanks for taking a stab at this @btkcodedev!

**Comment by hoechenberger** [comment]
Thanks for your contribution! I think the beginning of the sentence must also be adjusted, since we don't pass an "exclude" parameter anywhere as far as I can see

File: tutorials/preprocessing/15_handling_bad_channels.py
Comment (btkcodedev) [reply tutorials/preprocessing/15_handling_bad_channels.py]: Cool

**Comment by hoechenberger** [comment]
Thanks, but I don't think the sentence makes too much sense the way it's phrased now. I think it would be better to just write that "pick()" retains the bad channels. It's just how the methods works, not the justification why we used it.

File: tutorials/preprocessing/15_handling_bad_channels.py
Comment (mscheltienne) [inline tutorials/preprocessing/15_handling_bad_channels.py:242]:
```suggestion
# Note that the method :meth:`~mne.io.Raw.pick()` default
# arguments includes `exclude=()` which ensures that bad
# channels are not
```

**Comment by mscheltienne** [comment]
@btkcodedev Could you review the sentence modification I proposed and add a changelog entry?
For the chanbgelog entry, you can refer to [this section](https://mne.tools/dev/development/contributing.html#describe-your-changes-in-the-changelog) of the contributing guide. You need to add yourself to [the author list ](https://github.com/mne-tools/mne-python/blob/main/doc/changes/names.inc)(alphabetical order) and to add a file in [this folder](https://github.com/mne-tools/mne-python/tree/main/doc/changes/devel), named `12326.other.rst`, describe the change in the file and add yourself as author, e.g.

> Short description of the changes, by :newcontrib:`Firstname Lastname`.

**Comment by btkcodedev** [comment]
@mscheltienne Changes done :)

File: doc/changes/devel/12326.other.rst
Comment (mscheltienne) [inline doc/changes/devel/12326.other.rst:1]:
```suggestion
Updated the text in the preprocessing tutorial to use :class:`mne.io.Raw.pick()` instead of the legacy :class:`mne.io.Raw.pick_types()`, by `btkcodedev`_.
```

File: doc/changes/names.inc
Comment (mscheltienne) [inline doc/changes/names.inc:71]: Could you move this entry up, in its alphabetical position?

**Review by mscheltienne — COMMENTED** [review]
Thank you @btkcodedev 
2 more minor comments and this will be good to merge!

File: doc/changes/names.inc
Comment (drammock) [reply doc/changes/names.inc:71]: Am I correct that Balasubramanian is your surname? If so, to get correctly credited in our citations, it would be helpful if this entry were `T K Balasubramanian` (and sorted with the "T" entries)... the script that generates our citation file assumes that the surname comes last in these entries.

Comment (btkcodedev) [reply doc/changes/names.inc:71]: I could go with btkcodedev :)

Comment (drammock) [reply doc/changes/names.inc]: That is also ok if that's what you prefer!

**Comment by hoechenberger** [comment]
@wmvanvliet Could you please review again? Currently your change request (which AFAICT has been addressed) is blocking the merge. Thanks!

File: doc/changes/devel/12326.other.rst
Comment (drammock) [inline doc/changes/devel/12326.other.rst:1]: I'm a little surprised that these cross-refs worked; regardless, it's more accurate to mark these with the `:meth:` role since they're links to methods (not classes). The parentheses will get added automatically.

```suggestion
Updated the preprocessing tutorial to use :meth:`mne.io.Raw.pick` instead of the legacy :meth:`mne.io.Raw.pick_types`, by :newcontrib:`btkcodedev`.
```

File: tutorials/preprocessing/15_handling_bad_channels.py
Comment (drammock) [inline tutorials/preprocessing/15_handling_bad_channels.py:241]: FYI parentheses get added automatically when you use the `:meth:` role

```suggestion
# Note that the method :meth:`~mne.io.Raw.pick` default
```

**Review by drammock — APPROVED** [review]
just two small nitpicks from me. Thanks for fixing this!

**Review by wmvanvliet — APPROVED** [review]
Once @drammock's comments are resolved, LGTM

**Comment by wmvanvliet** [comment]
Didn't know there was an automerge in place. Perhaps we can fix @drammock's comments directly in main?

**Comment by hoechenberger** [comment]
Thanks for your contribution, @btkcodedev!!

**Comment by larsoner** [comment]
Somehow there is in `mne-tools/mne-python` :
```
$ git fetch upstream
...
 From github.com:mne-tools/mne-python
   03d78f43bf..990ce18d4e  main              -> upstream/main
 * [new branch]            revert-12326-main -> upstream/revert-12326-main
```
I'm going to delete the `revert-12326-main` assuming perhaps @wmvanvliet or someone else considered reverting for a minute there. Agree with @wmvanvliet that final comments can be addressed quickly elsewhere, but maybe just as part of another PR when someone opens one (rather than a push to `main` which is always a bit dicey)

**Comment by mscheltienne** [comment]
@wmvanvliet @larsoner I think I addressed the final comments before going on holidays in a PR, not yet merged, which was blocked by the towncrier job failing.

**Pull request merged**
